eclipse-collections/src/main/java/org/eclipse/collections/impl/bag/sorted/mutable/SynchronizedSortedBag.java

```diff
@@ -655,4 +656,22 @@ public int detectLastIndex(Predicate<? super T> predicate)
     {
         throw new UnsupportedOperationException("asParallel() method is not supported for " + this.getClass().getSimpleName() + '.');
     }
+
+    @Override
+    public Optional<T> getFirstOptional()
+    {
+        synchronized (this.getLock())
+        {
+            return this.getDelegate().getFirstOptional();
+        }
+    }
+
+    @Override
+    public Optional<T> getLastOptional()
+    {
+        synchronized (this.getLock())
+        {
+            return this.getDelegate().getLastOptional();
+        }
+    }
```

Comment:
I think these can be pulled up to `AbstractSynchronizedRichIterable`. It's hard to see in GitHub, but in general these methods should be right next to getFirst()/getLast() if they're at the right level of abstraction.